src/senselab/utils/tasks/dimensionality_reduction.py

```diff
@@ -0,0 +1,71 @@
+"""This module provides the implementation of dimensionality reduction."""
+
+import pydra
+import torch
+import umap
+from sklearn.decomposition import PCA as sklearn_PCA
+from sklearn.manifold import TSNE as sklearn_TSNE
+
+
+def compute_dimensionality_reduction(
+    data: torch.Tensor, model: str = "pca", n_components: int = 2, **kwargs: object
```

Comment:
how if I want to change some parameters (e.g., perplexity, random_state, max_iter, learning_rate, initial_state, ...)?